R/data-predictor.R

```diff
@@ -668,6 +669,66 @@ data_ac <- function(bterms, data, data2, basis = NULL, ...) {
       c(if (N_tg > 1L) begin_tg[2:N_tg], N + 1) - begin_tg
     ))
     out$end_tg <- with(out, begin_tg + nobs_tg - 1)
+    if (parameterize_ac_effects(acef)) {
```

Comment:
this adds a lot of new data variables to standata. Why can't we reuse all the variables that we already have for ARMA correlation structure? It seems they would also be applicable to the latent version.

Reply:
Thank you! I appreciate you looking into it!

Reply:
Upon reviewing this, I find that several of the existing variables (`begin_tg`, `end_tg`, and `nobs_tg`) are not used in the new formulation, so instead of defining new variables we could just change the definition of these. Furthermore, I think `ac_time` is no longer used anywhere, so I'll just remove it.

I do think `N_latent_err` and `latent_err_idx` are necessary to define the correct number of latent parameters and match them with rows in the data frame, and `max_time_span` is needed to avoid indexing the covariance matrix beyond its range.

Reply:
Changing the definition of old variables is possible but perhaps not ideal. Why would we need to change the meaning of them for use in the new structure? 

I agree that the three variables you mentioned at the end are likely needed.

Reply:
The old variables contain the start and end indices of groups in the vector of observations; the new ones contain indices in the vector of latent variables, which will be different indices if there are any time points observed more than once. I think one could extract the same information from the original definition of these variables combined with `latent_err_idx`, eliminating the need to change them, at the cost of adding a little more code to the transformed data block in the generated Stan program, and possibly something to prepare_predictions (though I would have to think about that as there's a lot there already that can probably be simplified).